File: perl/test-r_bin.pl
#!/usr/bin/perl

use r_bin;

local $file = ($ARGV[0] ne "")?$ARGV[0]:"/bin/ls";
local $b = r_bin::RBin->new ();
$b->load ($file, undef);
local $baddr = $b->get_baddr ();
local $info = $b->get_info ();
local @sects = $b->get_sections ();

print "$info\n";
print "  Type: $info->{type}\n";
print "  File: $info->{file}\n";
print "  Arch: $info->{arch}\n";
print "  VA: $info->{has_va}\n";
printf ("  Base Address: 0x%x\n", $baddr);

print "-> Sections $#sects\n";
print "$#sects\n";

# this fails because swig generates wrong glue for RList :(
for ($i = 0; $i < $#sects ; $i++) {
	$s = $sects[$i]; #->get ($i);
	$s = $sects->get ($i);
	printf ("offset=0x%08x va=0x%08x size=%05i %s\n",
			$s->{offset}, $baddr + $s->{rva}, $s->{size}, $s->{name});
}

File: python/test-r_bin.py
try:
	from r_bin import *
except:
	from r2.r_bin import *

path="/bin/ls"

io = RIO()
desc = io.open(path, 0, 0)
if desc == None:
	print "Cannot open file"
	exit(1)

print "FD", desc.fd

b = RBin ()
b.iobind(io)

# XXX broken test because now RBin depends on RIO
b.load (path, 0, 0, 0, desc.fd, False)
baddr = b.get_baddr ()
print 'Base Address', baddr
print '-> Sections'
for i in b.get_sections ():
	print 'offset=0x%08x va=0x%08x size=%05i %s' % (
			i.paddr, baddr+i.vaddr, i.size, i.name)

